### PR TITLE
analog: add commands

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/analog/utils/CommandBinaryBuffer.ts [hunk 1]
```diff
@@ -1750,13 +1712,59 @@ CommandBinaryBuffer.prototype.getExtendedValueSize = function ( bits: number ):
...
 
+CommandBinaryBuffer.prototype.getBatteryVoltage = function (): IBatteryVoltage {
+    let underHighLoad;
+    let underLowLoad;
+
+    const lowVoltageByte = this.getUint8();
+    const lowAndHightVoltageByte = this.getUint8();
+    const highVoltageByte = this.getUint8();
+
+    underLowLoad = lowVoltageByte << 4;
```
Comment (DarkPark) [inline src/analog/utils/CommandBinaryBuffer.ts:1734]:
```suggestion
    let underLowLoad = lowVoltageByte << 4;
```

File: src/analog/commands/downlink/correctTime2000.ts [hunk 2]
```diff
@@ -24,7 +24,6 @@
 import {TCommandId, TBytes, TUint8, TInt8} from '../../../types.js';
```
Comment (DarkPark) [inline src/analog/commands/downlink/correctTime2000.ts:24]:
```suggestion
import * as types from '../../../types.js';
```

?

to discuss

File: src/analog/commands/uplink/status.ts [hunk 3]
```diff
@@ -0,0 +1,305 @@
+/**
+ * Device status event.
+ *
+ * @packageDocumentation
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import Status from 'jooby-codec/analog/commands/uplink/Status.js';
+ *
+ * const commandBody = new Uint8Array([
```
Comment (DarkPark) [inline src/analog/commands/uplink/status.ts:10]: wrong example

File: src/analog/commands/downlink/getLmicInfo.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/analog/commands/GetLmicInfo.md#request)
+ */
+
+import {TCommandId, TBytes} from '../../../types.js';
+import * as command from '../../utils/command.js';
+
+
+export const id: TCommandId = 0x021f;
+export const headerSize = 3;
+const COMMAND_BODY_SIZE = 0;
```
Comment (DarkPark) [inline src/analog/commands/downlink/getLmicInfo.ts:23]:
```suggestion

const COMMAND_BODY_SIZE = 0;
```

and in other files

File: src/analog/commands/downlink/getArchiveEvents.ts [hunk 5]
```diff
@@ -1,24 +1,23 @@
-import {TBytes, TUint8} from '../../../types.js';
+import {TBytes, TUint8, TCommandId} from '../../../types.js';
```
Comment (DarkPark) [inline src/analog/commands/downlink/getArchiveEvents.ts:1]: would be nice to add description

File: src/analog/commands/downlink/setTime2000.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * // 240 seconds to the future
+ * const parameters = {sequenceNumber: 45, seconds: -120};
+ * const bytes = correctTime2000.toBytes(parameters);
+ *
+ * // command binary representation
+ * console.log(bytes);
+ * // [2, 5, 78, 43, 189, 152, 173]
+ * ```
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/analog/commands/SetTime2000.md#request)
```
Comment (DarkPark) [inline src/analog/commands/downlink/setTime2000.ts:16]:
```suggestion
 *
 * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/analog/commands/SetTime2000.md#request)
```

File: src/analog/commands/downlink/setTime2000.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+export const examples = {
+    'set time to 2023.04.03 14:01:17 GMT': {
+        id,
+        headerSize,
+        parameters: {
+            sequenceNumber: 78,
+            seconds: 733845677
+        },
+        bytes: [0x02, 0x05, 0x4e, 0x2b, 0xbd, 0x98, 0xad]
```
Comment (DarkPark) [inline src/analog/commands/downlink/setTime2000.ts:44]: would be nice to split it to two lines

and in some other places as well

File: src/analog/commands/uplink/getArchiveEvents.ts [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+ * ```js
+ * import * as getArchiveEvents from 'jooby-codec/analog/commands/uplink/getArchiveEvents.js';
+ *
+ * // 1 event "MAGNET_ON" at 2023.04.05 13:17:20 GMT
+ * const bytes = [0x2b, 0xc0, 0x31, 0x60, 0x01, 0x01];
+ *
+ * // decoded payload
+ * const parameters = getArchiveEvents.fromBytes(bytes);
+ *
+ * console.log(parameters);
```
Comment (DarkPark) [inline src/analog/commands/uplink/getArchiveEvents.ts:16]: such output is commented in other commands

shuold we uncomment it everywhere or comment here?

Comment (DarkPark) [reply src/analog/commands/uplink/getArchiveEvents.ts:16]: we can do it like this:

 * console.log(parameters);
 * // output:
 * {
 *     eventList: [
 *         {
 *             time2000: 734015840,
 *             id: 1,
 *             sequenceNumber: 1
 *         }
 *     ]
 * }

File: src/analog/commands/downlink/getArchiveEvents.ts [hunk 9]
```diff
@@ -1,24 +1,23 @@
...
 import CommandBinaryBuffer, {ICommandBinaryBuffer} from '../../utils/CommandBinaryBuffer.js';
 import * as command from '../../utils/command.js';
-import {ICommandParameters, TCommandExamples} from '../../utils/command.js';
 import {TTime2000} from '../../utils/time.js';
```
Comment (DarkPark) [inline src/analog/commands/downlink/getArchiveEvents.ts:5]:
```suggestion


```

and in other commands

File: src/analog/commands/uplink/setTime2000.ts [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * SetTime2000Response command parameters
+ *
+ * @example
+ * {status: 1}
+ */
+interface ISetTime2000ResponseParameters {
+    status: TUint8;
+}
+
```
Comment (DarkPark) [inline src/analog/commands/uplink/setTime2000.ts:36]:
```suggestion


```

and in other commands

File: src/analog/commands/uplink/status.ts [hunk 11]
```diff
@@ -0,0 +1,305 @@
...
+
+                if ( statusData.remainingBatteryCapacity === undefined ) {
+                    buffer.setUint8(UNKNOWN_BATTERY_CAPACITY);
+                } else {
+                    buffer.setUint8((UNKNOWN_BATTERY_CAPACITY - 1) * (statusData.remainingBatteryCapacity / 100));
+                }
+
+                buffer.setUint8(statusData.lastEventSequenceNumber);
+            }
+
```
Comment (DarkPark) [inline src/analog/commands/uplink/status.ts:277]:
```suggestion
```

?

File: src/analog/utils/CommandBinaryBuffer.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/analog/utils/CommandBinaryBuffer.ts]: fixed

File: src/analog/commands/downlink/correctTime2000.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/correctTime2000.ts]: done for all files

File: src/analog/commands/uplink/status.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/analog/commands/uplink/status.ts]: fixed

File: src/analog/commands/downlink/getLmicInfo.ts (hunk 4 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/getLmicInfo.ts:24]: fixed

File: src/analog/commands/downlink/getArchiveEvents.ts (hunk 5 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/getArchiveEvents.ts]: description added

File: src/analog/commands/downlink/setTime2000.ts (hunk 6 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/setTime2000.ts:17]: fixed

File: src/analog/commands/downlink/setTime2000.ts (hunk 7 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/setTime2000.ts]: fixed

File: src/analog/commands/uplink/getArchiveEvents.ts (hunk 8 above)
Comment (DeadByElpy) [reply src/analog/commands/uplink/getArchiveEvents.ts:16]: comments removed, console - already output, so 'output:' comment no necessary to add

File: src/analog/commands/uplink/status.ts (hunk 11 above)
Comment (DeadByElpy) [reply src/analog/commands/uplink/status.ts]: fixed

File: src/analog/commands/downlink/getArchiveEvents.ts (hunk 9 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/getArchiveEvents.ts:26]: fixed

File: src/analog/commands/uplink/time2000.ts [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+ * It is sent immediately after device power on. After it a device sends it periodically (once per 24 hours).
+ *
+ * @packageDocumentation
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import Time2000 from 'jooby-codec/analog/commands/uplink/Time2000.js';
+ *
+ * // failure
+ * const commandBody = new Uint8Array([0x4d, 0x2b, 0xbd, 0x98, 0xad]);
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/analog/commands/uplink/time2000.ts:14]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**